packages/components/pie-toast/src/index.ts

```diff
@@ -1,24 +1,259 @@
+     */
+    protected async updated (_changedProperties: PropertyValues<this>) {
+        if (_changedProperties.has('isOpen') && this.isOpen) {
+            dispatchCustomEvent(this, ON_TOAST_OPEN_EVENT, { targetNotification: this });
+        }
+
+        await this.updateComplete;
+
+        if (this.actionButton) {
+            this._actionButtonOffset = this.actionButton.offsetWidth;
+        }
+
+        // Temporary const. This will be removed when we implement variants.
+        const hasIcon = true;
+
+        this._messageAreaMaxWidth = this.getMessageMaxWidth(hasIcon, this.isMultiline, !!this.leadingAction?.text, this.isDismissible);
+
+        if (
+            _changedProperties.has('message') ||
+            _changedProperties.has('isDismissible') ||
+            _changedProperties.has('isMultiline') ||
+            _changedProperties.has('leadingAction')) {
+            this.requestUpdate();
+        }
+    }
+
+    /**
+     * It handle the action button action.
+     *
+     * @private
+     */
+    private handleActionClick () {
+        dispatchCustomEvent(this, ON_TOAST_LEADING_ACTION_CLICK_EVENT, { targetNotification: this });
+    }
+
+    /**
+     * Render the action button depending on action type and its action.
+     *
+     * @param {ActionProps} action - The action properties.
+     *
+     * @returns {TemplateResult | typeof nothing} - The rendered action button or nothing if the action text is not defined.
+     * @private
+     */
+    private renderActionButton (action: ActionProps) : TemplateResult | typeof nothing {
+        const { text, ariaLabel } = action;
+
+        return html`
+            <pie-button
+                variant="ghost-inverse"
+                size="xsmall"
+                aria-label="${ariaLabel || nothing}"
+                @click="${() => this.handleActionClick()}"
+                class="${componentSelector}-leading-action"
+                data-test-id="${componentSelector}-leading-action"
+                type="button">
+                ${text}
+            </pie-button>
+        `;
+    }
+
+    /**
+     * Template for the footer area.
+     * It should display only when isMultiline is true as well if has action button.
+     * Called within the main render function.
+     *
+     * @private
+     */
+    private renderFooter () {
+        const { leadingAction } = this;
+        return html`
+            <footer class="${componentClass}-footer" data-test-id="${componentSelector}-footer" >
+                ${leadingAction ? this.renderActionButton(leadingAction) : nothing}
+            </footer>
+        `;
+    }
+
+    /**
+     * Template for the close button element. Called within the
+     * main render function.
+     *
+     * @private
+     */
+    private renderCloseButton (): TemplateResult {
+        return html`
+            <pie-icon-button
+                variant="ghost-inverse"
+                size="xsmall"
+                class="${componentClass}-icon-close"
```

Comment:
I would just change this class to:

```suggestion
                class="${componentClass}-closeBtn"
```